src/components/menu-surface/menu-surface.tsx

```diff
@@ -30,6 +30,9 @@ export class MenuSurface {
     @Prop()
     public open = false;
 
+    @Prop()
+    public cssClasses: string[] = [];
```

Comment:
I'm not sure about this :thinking: It makes it a bit unclear for the consumer what we actually can do to change the style of the menu. Maybe we should have one or more properties with explicit names that can be set, and set the css classes internally inside the menu instead. What do you think?

Reply:
Or if it can be solved with css properties it would be even better I guess :slightly_smiling_face:, but that might not be possible here

Reply:
agree, I didn't like the solution too much either, but I also didn't x properties for x different styling combinations. I'll look into doing it as css variables but otherwise we could have one property as object with boolean like this 
```
{dropdown-list: boolean;
list-full: boolean}
```
probably different names but I guess you get the idea

Reply:
Does it have to be styled inside the menu surface? Would it be possible to have the styling on the components that are affected? :thinking:

Reply:
We can always add styling as a `style` attribute if don't find a better solution

Reply:
I should be able to wrap the content that I render inside the menu surface with a div and set the css classes on that. that shouldn't need any changes in the surface component.

Reply:
shit, it doesn't work that way. it has to be inside a shadow dom if we don't want to use purely inline style. I'll look into a combination of concreter props and inline styling

Reply:
If we want to send in styling it might make more sense to do it on the portal than on the menu surface, since that is the component that moves its children to somewhere else in the DOM :thinking: 
That means we could do something like this

```tsx
<limel-portal styling={{}}>
    <my-component />
</limel-portal>
```
And inside the portal
```tsx
render() {
    return [
        <style>{this.styling}</style>,
        <slot />
    ];
}
```

What do you think? Does it make sense?

Reply:
I think that makes to our own styles on the dropdown list. but not too much for using material's elevation classes. My current idea is to have a prop on the surface like this
```
    /**
     * use material design's elevation styling
     */
    @Prop()
    public elevationDesign: boolean;
```
and use the styling you suggested for the rest